clustering.py

```diff
@@ -263,10 +267,10 @@ def kmeans_constrained(data, label_dict, ts_to_labels):
         series was placed in.
     """
     must_link, can_not_link = make_constraints(ts_to_labels)
-    num_clusters = (len(data) // 20) + 6
+    num_clusters = (len(data) // params["kmeans_ratio"]) + params["kmeans_min"]
     clusters_distances = []
 
-    for run_num in range(5):
+    for run_num in range(10):
```

Comment:
Even if this is just convention, it's still worth commenting where the number 10 came from. No "magic numbers."